### DISCUSSION TITLE
Getting uWSGI error. Exiting due to SIGTERM

### DISCUSSION THREAD

**Discussion by pupattan** [tiangolo/uwsgi-nginx-docker#223] [Questions] [answered]
*** Starting uWSGI 2.0.19.1 (64bit) on [Fri Dec 22 18:40:09 2023] ***
compiled with version: 10.2.1 20201203 on 09 January 2021 08:06:39
os: Linux-6.5.0-14-generic #14-Ubuntu SMP PREEMPT_DYNAMIC Tue Nov 14 14:59:49 UTC 2023
nodename: a358b004ca18
machine: x86_64
clock source: unix
pcre jit disabled
detected number of CPU cores: 1
current working directory: /usr/src/app
detected binary path: /usr/sbin/uwsgi
your memory page size is 4096 bytes
detected max file descriptor number: 1048576
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to UNIX address /tmp/uwsgi.sock fd 3
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) ***
Python version: 3.8.15 (default, Nov 10 2022, 13:35:13)  [GCC 10.2.1 20201203]
*** Python threads support is disabled. You can enable it with --enable-threads ***
Python main interpreter initialized at 0x7f9f93607450
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) ***
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 364600 bytes (356 KB) for 4 cores
*** Operational MODE: preforking ***
running here
WSGI app 0 (mountpoint='') ready in 1 seconds on interpreter 0x7f9f93607450 pid: 14 (default app)
2023-12-22 18:40:11,083 INFO success: quit_on_failure entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-12-22 18:40:11,085 WARN received SIGTERM indicating exit request
2023-12-22 18:40:11,090 INFO waiting for quit_on_failure, uwsgi to die
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) ***
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 14)
spawned uWSGI worker 1 (pid: 16, cores: 1)
spawned uWSGI worker 2 (pid: 17, cores: 1)
running "unix_signal:15 gracefully_kill_them_all" (master-start)...
2023-12-22 18:40:14,109 INFO waiting for quit_on_failure, uwsgi to die
2023-12-22 18:40:17,113 INFO waiting for quit_on_failure, uwsgi to die
2023-12-22 18:40:20,117 INFO waiting for quit_on_failure, uwsgi to die
2023-12-22 18:40:21,117 WARN killing 'uwsgi' (14) with SIGKILL
Fri Dec 22 18:40:21 2023 - uWSGI worker 1 screams: UAAAAAAH my master disconnected: i will kill myself !!!
Fri Dec 22 18:40:21 2023 - uWSGI worker 2 screams: UAAAAAAH my master disconnected: i will kill myself !!!
2023-12-22 18:40:21,130 INFO stopped: uwsgi (terminated by SIGKILL)
2023-12-22 18:40:21,131 INFO stopped: quit_on_failure (terminated by SIGTERM)

**Comment by pupattan** [comment]
Using with  https://github.com/nginx-proxy/nginx-proxy and jrcs/letsencrypt-nginx-proxy-companion

**Comment by tiangolo** [comment, marked as answer]
That seems like an error in the app code, but it's hard to know without a code example. :thinking:
